### PR TITLE
Rewrite Starting and stopping instances to use tt CLI

### LINKED ISSUE #3718: Rewrite Starting and stopping an instance for tt
**Product:** Tarantool, tt
**Since:** 2.11
**Audience/target:** devs
**Root document:** https://www.tarantool.io/en/doc/latest/book/admin/start_stop_instance/
**SME:** @ psergee 

# Details

Follow-up on #3501 

The page Starting and stopping an instance explains in details how to manage Tarantool instances with `tarantoolctl`.
In scope of `tarantoolctl` deprecation this page needs to be rewritten. It should generally reflect the `tt` CLI approach and
explain its specifics.

# Definition of done

- [ ] the page Starting and stopping an instance shows how to manage instances with `tt`
- [ ] The current content (about `tarantoolctl`) is moved to the `tarantoolctl` section.

### PR DESCRIPTION
**Pull request by p7nov**
Resolves #3718 

- Rewrote **Starting and stopping instances** to show the basic instance management scenarios with `tt` CLI.
- Moved the old content about `tarantoolctl` to the tarantoolctl doc page.
- Moved the content about multi-instance application from tt reference to  **Starting and stopping instances** (redirect: https://github.com/tarantool/tarantool.io/pull/1449)

Deployment: https://docs.d.tarantool.io/en/doc/gh-3718-starting-instance-tt/book/admin/start_stop_instance/

### CODE REVIEW

File: doc/book/admin/start_stop_instance.rst [hunk 1]
```diff
@@ -1,214 +1,199 @@
...
 
-    $ systemctl enable tarantool@my_app
+    .. code-block:: console
 
-To stop a running ``my_app`` instance, say:
+        $ tt check my_app
+           • Result of check: syntax of file '/etc/tarantool/instances.enabled/my_app.lua' is OK
 
-.. code-block:: console
+*   ``tt status`` -- check the instance's status:
```
Comment (psergee) [inline doc/book/admin/start_stop_instance.rst]: instance's -> instance
?

**Review by psergee — APPROVED** [review]
Shouldn't we add an example of multi-instance application?

**Comment by p7nov** [comment]
> Shouldn't we add an example of multi-instance application?

Thanks for the idea.
Added here the content about multi-instance applications from tt reference section.

File: doc/book/admin/start_stop_instance.rst [hunk 2]
```diff
@@ -1,214 +1,335 @@
...
-To enable ``my_app`` instance for auto-load during system startup, say:
+        $ tt restart my_app -y
+           • The Instance my_app (PID = 729) has been terminated.
+           • Starting an instance [my_app]...
 
-.. code-block:: console
+    .. note::
 
-    $ systemctl enable tarantool@my_app
+        The ``-y`` option responds "yes" to confirmation prompt automatically.
```
Comment (andreyaksenov) [inline doc/book/admin/start_stop_instance.rst]:
```suggestion
        The ``-y`` option responds "yes" to the confirmation prompt automatically.
```

File: doc/book/admin/start_stop_instance.rst [hunk 3]
```diff
@@ -1,214 +1,335 @@
...
 
-   default_cfg = {
-       pid_file  = "/home/user/tarantool_test/my_app.pid",
-       wal_dir   = "/home/user/tarantool_test",
-       snap_dir  = "/home/user/tarantool_test",
-       vinyl_dir = "/home/user/tarantool_test",
-       log       = "/home/user/tarantool_test/log",
-   }
-   instance_dir = "/home/user/tarantool_test"
+To create an multi-instance application, prepare its layout
```
Comment (andreyaksenov) [inline doc/book/admin/start_stop_instance.rst]:
```suggestion
To create a multi-instance application, prepare its layout
```

File: doc/book/admin/start_stop_instance.rst [hunk 4]
```diff
@@ -1,214 +1,335 @@
...
 
-================================================================================
-Starting/stopping an instance
-================================================================================
+Starting and stopping instances
+===============================
 
-While a Lua application is executed by Tarantool, an instance file is executed
-by ``tarantoolctl`` which is a Tarantool script.
+Quick start
```
Comment (andreyaksenov) [inline doc/book/admin/start_stop_instance.rst]: Honestly, I'd remove this heading as some people might call this not a "quick" start. For example, the `instance file` link leads us to a huge section of text without any kind of `hello world` instance file example, so starting an instance might be not very simple. I think, the text below works good as an introduction without a heading.

File: doc/book/admin/start_stop_instance.rst [hunk 5]
```diff
@@ -1,214 +1,335 @@
...
+           • Starting an instance [my_app]...
+
+In this case, ``tt`` starts an instance from any ``*.lua`` file it finds in ``/etc/tarantool/instances.enabled/``.
+
+Starting instances
+------------------
+
+All the instance files or directories placed in the ``instances_enabled`` directory
+specified in :ref:`tt configuration <tt-config_file>` are called *enabled instances*.
+If there are several enabled instances, ``tt start`` starts a separate instance for each of them.
```
Comment (andreyaksenov) [inline doc/book/admin/start_stop_instance.rst]: The unfolded sentence sounds a bit weird: `.. starts a separate instance for each of the instances`. Looks like the `enabled instance` term adds some confusion with a Tarantool instance. Maybe smth like this?

```
If there are several enabled instances, ``tt start`` starts them separately.
```

or use `Tarantool instance` to add some contrast:

```
If there are several enabled instances, ``tt start`` starts a separate Tarantool instance...
```

File: doc/book/admin/start_stop_instance.rst [hunk 6]
```diff
@@ -1,214 +1,335 @@
...
 
-   When ``tarantool`` is invoked by root, it looks for a configuration file in
-   ``/etc/default/tarantool``. When ``tarantool`` is invoked by a local (non-root)
-   user, it looks for a configuration file first in the current directory
-   (``$PWD/.tarantoolctl``), and then in the current user's home directory
-   (``$HOME/.config/tarantool/tarantool``). If no configuration file is found
-   there, or in the ``/usr/local/etc/default/tarantool`` file,
-   then ``tarantoolctl`` falls back to
-   :ref:`built-in defaults <admin-tt_config_file>`.
+*   ``/var/log/tarantool/<instance_name>.log`` -- instance :ref:`logs <admin-logs>`
```
Comment (andreyaksenov) [inline doc/book/admin/start_stop_instance.rst]:
```suggestion
*   ``/var/log/tarantool/<instance_name>.log`` -- instance :ref:`logs <admin-logs>`.
```

File: doc/book/admin/start_stop_instance.rst [hunk 7]
```diff
@@ -1,214 +1,335 @@
...
+*   ``/var/lib/tarantool/<instance_name>/`` -- snapshots and write-ahead logs
```
Comment (andreyaksenov) [inline doc/book/admin/start_stop_instance.rst]: I'd add the dot for consistency with the next bullets containing several sentences.
```suggestion
*   ``/var/lib/tarantool/<instance_name>/`` -- snapshots and write-ahead logs.
```

File: doc/book/admin/start_stop_instance.rst [hunk 8]
```diff
@@ -1,214 +1,335 @@
...
+*   ``/var/run/tarantool/<instance_name>.control`` -- control socket. This is
```
Comment (andreyaksenov) [inline doc/book/admin/start_stop_instance.rst]:
```suggestion
*   ``/var/run/tarantool/<instance_name>.control`` -- a control socket. This is
```

File: doc/book/admin/start_stop_instance.rst [hunk 9]
```diff
@@ -1,214 +1,335 @@
...
+    a Unix socket with Lua console attached to it. This file is used to connect
```
Comment (andreyaksenov) [inline doc/book/admin/start_stop_instance.rst]:
```suggestion
    a Unix socket with the Lua console attached to it. This file is used to connect
```

File: doc/book/admin/start_stop_instance.rst [hunk 10]
```diff
@@ -1,214 +1,335 @@
...
+    to the instance console.
+*   ``/var/run/tarantool/<instance_name>.pid`` -- PID file that ``tt`` uses to
```
Comment (andreyaksenov) [inline doc/book/admin/start_stop_instance.rst]:
```suggestion
*   ``/var/run/tarantool/<instance_name>.pid`` -- a PID file that ``tt`` uses to
```

File: doc/book/admin/start_stop_instance.rst [hunk 11]
```diff
@@ -1,214 +1,335 @@
...
-    $ tarantoolctl check my_app
+    .. code-block:: console
 
-To enable ``my_app`` instance for auto-load during system startup, say:
+        $ tt restart my_app -y
+           • The Instance my_app (PID = 729) has been terminated.
+           • Starting an instance [my_app]...
 
-.. code-block:: console
+    .. note::
```
Comment (andreyaksenov) [inline doc/book/admin/start_stop_instance.rst]: Maybe, it's better to format this as a regular sentence. A note takes too much space and attention.

File: doc/book/admin/start_stop_instance.rst [hunk 12]
```diff
@@ -1,214 +1,335 @@
...
 
-... and set default ``tarantoolctl`` configuration in
-``$HOME/.config/tarantool/tarantool``. Let the file contents be:
+*   start an application on multiple instances
+*   check the status of application instances
+*   connect to a specific instance of an application
+*   stop a specific instance of an application or all its instances
 
-.. code-block:: lua
+Application layout
```
Comment (andreyaksenov) [inline doc/book/admin/start_stop_instance.rst:127]: Future task: we need to think about a new place for this section as it not directly related to administration. Ideally, this section should contain only the `Managing multi-instance applications` content.

File: doc/book/admin/start_stop_instance.rst [hunk 13]
```diff
@@ -1,214 +1,335 @@
...
 
-Let’s verify our instance file by starting it without ``tarantoolctl`` first:
+*   (Optional) Specific instances files.
+    These files should have names ``<instance_name>.init.lua``, where ``<instance_name>``
+    is the name specified in ``instances.yml``.
+    For example, if your application has separate source files for the ``router`` and ``storage``
+    instances, place the router code in the ``router.init.lua`` file.
 
-.. code-block:: console
+Example: a ``demo`` application that has three instances (``storage1``, ``storage2``, and ``router``).
```
Comment (andreyaksenov) [inline doc/book/admin/start_stop_instance.rst]:
```suggestion
Example: a ``demo`` application that has three instances:``storage1``, ``storage2``, and ``router``.
```

File: doc/book/admin/start_stop_instance.rst [hunk 14]
```diff
@@ -1,214 +1,335 @@
...
+
+.. _admin-start_stop_instance-running_locally:
+
+Running Tarantool locally
+-------------------------
+
+Sometimes you may need to run a Tarantool instance locally, for example, for test
+purposes. ``tt`` runs in a local environment if it finds a ``tt.yaml`` configuration
+file in the current directory or any of its enclosing directories.
+To force ``tt`` into the local mode, add the ``-L`` or ``--local`` argument.
```
Comment (andreyaksenov) [inline doc/book/admin/start_stop_instance.rst]: Let's add a link to https://www.tarantool.io/en/doc/latest/reference/tooling/tt_cli/configuration/#launch-modes somewhere. And I didn't quite understand which `tt` command should be run with the `-L` flag.

**Review by andreyaksenov — CHANGES_REQUESTED** [review]

File: doc/book/admin/start_stop_instance.rst (hunk 14 above)
Comment (p7nov) [reply doc/book/admin/start_stop_instance.rst]: moved the part about --local in the end of the section with a separate example.

**Review by andreyaksenov — APPROVED** [review]

**Pull request merged**